tools/windows/install-help/cal/customactiondata.cpp

```diff
@@ -252,9 +252,22 @@ void CustomActionData::findSuppliedUserInfo(std::wstring &input, std::wstring &c
 
     if (computed_domain == L".")
     {
-        WcaLog(LOGMSG_STANDARD, "Supplied qualified domain '.', using hostname");
-        computed_domain = machine.GetMachineName();
-        domainUser = false;
+        if (GetTargetMachine().IsDomainController())
+        {
+            // User didn't specify a domain OR didn't specify a user, but we're on a domain controller
+            // let's use the joined domain.
+            computed_domain = machine.JoinedDomainName();
+            domainUser = true;
```

Comment:
To make sure I understand the code, let me know if this is correct:
- Previous behavior: when installing on a DC, we can guess the domain name if not specified but you still need to specify the username and password.
- New behavior: when installing on a DC, just specifying the password is enough: we guess the domain name and assume the user is `ddagentuser`.

Reply:
Aren't we saying the same thing? On DC:
Old behavior: need to specify user and pass.
New behavior: only need to specify pass.

Reply:
Make sure you do need to specify at least the pass at all times on a DC.